lib/collection.js

```diff
@@ -1838,19 +1839,37 @@ Collection.prototype.count = function(query, options, callback) {
   return executeOperation(this.s.topology, count, [this, query, options, callback]);
 };
 
-var count = function(self, query, options, callback) {
-  var skip = options.skip;
-  var limit = options.limit;
-  var hint = options.hint;
-  var maxTimeMS = options.maxTimeMS;
+/**
+ * Gets an estimate of the count of documents in a collection using collection metadata.
+ * @method
+ * @param {object} [options] Optional settings.
+ * @param {number} [options.maxTimeMS] The maximum amount of time to allow the operation to run.
+ * @param {Collection~countCallback} [callback] The command result callback.
+ * @return {Promise} returns Promise if no callback passed.
+ */
+Collection.prototype.estimatedDocumentCount = function(options, callback) {
+  const args = Array.prototype.slice.call(arguments, 0);
```

Comment:
let's not use this pattern where not needed. Since in this function we aren't dealing with positional parameters other than `options` we can simply put:
```
if (typeof options === 'function') (callback = options), (options = {});
options = options || {};
```

and then `executeOperation`